<img src="Tardigrade.png" alt="Tardigrade" style="width:200px">

<H3><center><b>Picard-Fuchs equation for the tardigrade with 5 points in four dimensions</b></center></H3>
$$
U=x_3 x_4 + x_3 x_5 + x_4 x_5 + x_3 x_6 + x_4 x_6 + x_2 (x_3 + x_5 + x_6) + 
 x_1 (x_2 + x_4 + x_5 + x_6)
$$
$$L=m_1^2 x_1 + \cdots + m_6^2 x_6
$$
$$V_{5pt}=p_1^2  x_1  x_2  x_5 + s_{12}  x_2  x_3  x_5 + p_1^2  x_1  x_4  x_5 + p_5^2  x_1  x_4  x_5 - 
 s_{45}  x_1  x_4  x_5 - s_{51}  x_1  x_4  x_5 + p_3^2  x_3  x_4  x_5 + p_5^2  x_3  x_4  x_5 + 
 s_{12}  x_3  x_4  x_5 - s_{23}  x_3  x_4  x_5 - s_{34}  x_3  x_4  x_5 - s_{45}  x_3  x_4  x_5 - 
 s_{51}  x_3  x_4  x_5 + s_{34}  x_2  x_3  x_6 + s_{23}  x_1  x_4  x_6 + p_3^2  x_3  x_4  x_6 + 
 p_5^2  x_1  x_5  x_6 - s_{23}  x_1  x_5  x_6 - s_{45}  x_1  x_5  x_6 - s_{51}  x_1  x_5  x_6 + 
 p_5^2  x_2  x_5  x_6 - s_{23}  x_2  x_5  x_6 - s_{45}  x_2  x_5  x_6 - s_{51}  x_2  x_5  x_6 + 
 p_5^2  x_3  x_5  x_6 - s_{23}  x_3  x_5  x_6 - s_{45}  x_3  x_5  x_6 - s_{51}  x_3  x_5  x_6 + 
 p_5^2  x_4  x_5  x_6 - s_{23}  x_4  x_5  x_6 - s_{45}  x_4  x_5  x_6 - s_{51}  x_4  x_5  x_6 + 
 p_2^2  x_1  x_3  (x_2 + x_4 + x_5 + x_6) + 
 p_4^2  x_4  (x_3  x_5 + x_1  (x_2 + x_5) + x_2  (x_3 + x_5 + x_6)) $$
 
 $$
 V_{3pt}=p_3^2  ((x_2 + x_4)  x_5  x_6 + x_1  (x_2 + x_5)  x_6 + x_3  x_5  (x_4 + x_6)) + 
 p_2^2  (x_1  x_4  (x_2 + x_5) + x_2  x_3  (x_4 + x_6) + x_2  x_4  (x_5 + x_6)) + 
 p_1^2  (x_2  x_3  x_5 + x_1  (x_2  x_3 + x_4  x_6 + x_3  (x_4 + x_5 + x_6)))
 $$
 
$$Int =\int {dx_1\cdots dx_6\over
     \left(
      UL-tV \right)^{2}}
$$

In [2]:
from ore_algebra import *
OA, z, Dz = DifferentialOperators(QQ, 'z')

``F := (x1 + 841*x2 + 2809*x3 + x4 + 121*x5 + 3481*x6)*(x3*x4 + x3*x5 + x4*x5 + x3*x6 + x4*x6 + x2*(x3 + x5 + x6) + x1*(x2 + x4 + x5 + x6)) - t*(5329*x1*x2*x5 + 7*x2*x3*x5 + 11524*x1*x4*x5 + 6623*x3*x4*x5 + 37*x2*x3*x6 + 71*x1*x4*x6 + 529*x3*x4*x6 + 6124*x1*x5*x6 + 6124*x2*x5*x6 + 6124*x3*x5*x6 + 6124*x4*x5*x6 + 1681*x1*x3*(x2 + x4 + x5 + x6) + 841*x4*(x3*x5 + x1*(x2 + x5) + x2*(x3 + x5 + x6)))``
<br>
The Picard-Fuchs differential operator is given in <a href="PF-Tardigrade-5pt-d4-case1.txt">PF-Tardigrade-5pt-d4-case1.txt</a>

In [4]:
s = open("PF-Tardigrade-5pt-d4-case1.txt").readlines()[0]
dopTardigrade5ptd4case1 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopTardigrade5ptd4case1.order()]

[11]

In [5]:
dopTardigrade5ptd4case1.leading_coefficient().factor()

(z - 2916/1681) * (z - 2704/1681) * (z - 900/841) * (z - 784/841) * (z - 1225/1531) * (z - 576/1531) * z^3 * (z^24 + 3817616875663137549249648305618044339937347067692394912864038835714420707750597889950576572221881310150638137169403339745475929400475785440334282363/35692654361990926180940633153227036669725100870087153503083153087692836180687918959441381963612630214153976950130074613902529271194998451879561875*z^23 + 3699124390749271140026132523254286568600374465005540625691384232815267088441233220732379944588027194549999955752487090928611715222261285482982546965989881430797859/927041107577618335510358064253467971116574791318592352109941913988210140957219268264972619059412114547733578557091426025661770409629899274190210589540867500000*z^22 + 242284920664826520776096404975188730306175265482015411576685388308510624145537992048398588161460148567281021649467532338807134035410084526521653504514604547299701568441340673/40129882487992502792836711514666085543875691959677132211659251676962695246

`Q= -(t*(-6124*x1^4 - 4331388844*x2^4 - 48321305644*x3^4 + 11236*(2735842 + 2861439*t)*x3^3*x4 - 3*(3530775692 + t*(-3975638889 + 8060114848*t))*x3^2*x4^2 + 4*(2735842 + 2861439*t)*x3*x4^3 - 6124*x4^4 - 6728*x2^3*((2843419 + 4654240*t)*x3 - 1531*(-842 + 841*t)*x4) + 4*x2^2*(-((8255351409 + 7*t*(4847377523 + 56647*t))*x3^2) + (-2487474474 + t*(-648639203 + 7828431680*t))*x3*x4 - 1531*(710646 + 841*t*(-1684 + 841*t))*x4^2) + 4*x2*(-5618*(2843419 + 4654240*t)*x3^3 + 3*(-845005726 + t*(7442039872 + 2485534101*t))*x3^2*x4 + 3*(765964042 + (33060012 - 802156733*t)*t)*x3*x4^2 + 3062*(-842 + 841*t)*x4^3) + 2*x1*(2*(-841*(-7412818 + 1217145*t)*x2^3 + 3*(4099527954 + t*(688473482 + 5152973967*t))*x2^2*x3 + (4842590982 + t*(-20016214427 + 15647554880*t))*x2*x3^2 + 8601158*(-2810 + 1681*t)*x3^3) + (2*(19559246574 + t*(868132124 + 1023618945*t))*x2^2 + (38164345848 + t*(115968824921 + 75891581202*t))*x2*x3 + 2*(52169629902 + (30586947836 - 4810078959*t)*t)*x3^2)*x4 - 2*((-13340902134 + t*(5182059139 + 8127148152*t))*x2 + (9953242938 + 5*(7971225438 - 987009673*t)*t)*x3)*x4^2 + 4*(7917749 + 4831836*t)*x4^3) + 4*x1^3*((7412818 - 1217145*t)*x2 + 15835498*x4 + 3062*(-2810*x3 + 1681*t*x3 + 3156*t*x4)) + x1^2*(3*(-11870168524 + 3549367865*t)*x2^2 - 6124*(7901718 + 1681*t*(-5620 + 1681*t))*x3^2 + 4*(44500485222 + (538307621 - 16244632632*t)*t)*x3*x4 - 4*(36761722833 + t*(66334271909 + 1344087865*t))*x4^2 + 4*x2*(3*(6941443914 + t*(-5296810996 + 682006915*t))*x3 + (-37295933226 + t*(-21173159916 + 7305817175*t))*x4))))`

In [6]:
s = open("PF-Quartic-case1.txt").readlines()[0]
dopQuarticcase1 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopQuarticcase1.order()]

[11]

In [7]:
dopQuarticcase1.leading_coefficient().factor()

(z - 2916/1681) * (z - 2704/1681) * (z - 900/841) * (z - 784/841) * z^3 * (z - 1225/1531)^10 * (z - 576/1531)^10 * (z^24 + 3817616875663137549249648305618044339937347067692394912864038835714420707750597889950576572221881310150638137169403339745475929400475785440334282363/35692654361990926180940633153227036669725100870087153503083153087692836180687918959441381963612630214153976950130074613902529271194998451879561875*z^23 + 3699124390749271140026132523254286568600374465005540625691384232815267088441233220732379944588027194549999955752487090928611715222261285482982546965989881430797859/927041107577618335510358064253467971116574791318592352109941913988210140957219268264972619059412114547733578557091426025661770409629899274190210589540867500000*z^22 + 242284920664826520776096404975188730306175265482015411576685388308510624145537992048398588161460148567281021649467532338807134035410084526521653504514604547299701568441340673/40129882487992502792836711514666085543875691959677132211659251676962

In [9]:
dop=dopQuarticcase1.annihilator_of_composition(z)

In [12]:
dop-dopQuarticcase1

0

In [11]:
dop.leading_coefficient().factor()

(z - 2916/1681) * (z - 2704/1681) * (z - 900/841) * (z - 784/841) * z^3 * (z - 1225/1531)^10 * (z - 576/1531)^10 * (z^24 + 3817616875663137549249648305618044339937347067692394912864038835714420707750597889950576572221881310150638137169403339745475929400475785440334282363/35692654361990926180940633153227036669725100870087153503083153087692836180687918959441381963612630214153976950130074613902529271194998451879561875*z^23 + 3699124390749271140026132523254286568600374465005540625691384232815267088441233220732379944588027194549999955752487090928611715222261285482982546965989881430797859/927041107577618335510358064253467971116574791318592352109941913988210140957219268264972619059412114547733578557091426025661770409629899274190210589540867500000*z^22 + 242284920664826520776096404975188730306175265482015411576685388308510624145537992048398588161460148567281021649467532338807134035410084526521653504514604547299701568441340673/40129882487992502792836711514666085543875691959677132211659251676962

``F := (6889*x1 + 841*x2 + 2809*x3 + 529*x4 + 121*x5 + 3481*x6)*(x3*x4 + x3*x5 + x4*x5 + x3*x6 + x4*x6 + x2*(x3 + x5 + x6) + x1*(x2 + x4 + x5 + x6)) - t*(5329*x1*x2*x5 + 7*x2*x3*x5 + 11524*x1*x4*x5 + 6623*x3*x4*x5 + 37*x2*x3*x6 + 71*x1*x4*x6 + 529*x3*x4*x6 + 6124*x1*x5*x6 + 6124*x2*x5*x6 + 6124*x3*x5*x6 + 6124*x4*x5*x6 + 1681*x1*x3*(x2 + x4 + x5 + x6) + 841*x4*(x3*x5 + x1*(x2 + x5) + x2*(x3 + x5 + x6)))``
<br>
The Picard-Fuchs differential operator is given in <a href="PF-Tardigrade-5pt-d4-case2.txt">PF-Tardigrade-5pt-d4-case2.txt</a>

In [9]:
s = open("PF-Tardigrade-5pt-d4-case2.txt").readlines()[0]
dopTardigrade5ptd4case2 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopTardigrade5ptd4case2.order()]

[11]

``F := (x1 + x2 + x3 + x4 + x5 + x6)*(x3*x4 + x3*x5 + x4*x5 + x3*x6 + x4*x6 + x2*(x3 + x5 + x6) + x1*(x2 + x4 + x5 + x6)) - t*(x1*x2*x5 + x2*x3*x5 + x1*x4*x5 + x3*x4*x5 + x2*x3*x6 + x1*x4*x6 + x3*x4*x6 + x1*x5*x6 + x2*x5*x6 + x3*x5*x6 + x4*x5*x6 + x1*x3*(x2 + x4 + x5 + x6) + x4*(x3*x5 + x1*(x2 + x5) + x2*(x3 + x5 + x6)))``
<br>
The Picard-Fuchs differential operator is given in <a href="PF-Tardigrade-5pt-d4-case3.txt">PF-Tardigrade-5pt-d4-case3.txt</a>

In [3]:
s = open("PF-Tardigrade-5pt-d4-case3.txt").readlines()[0]
dopTardigrade5ptd4case3 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopTardigrade5ptd4case3.order(),dopTardigrade5ptd4case3.degree(),dopTardigrade5ptd4case3.indicial_polynomial(z).factor()]

[6, 34, (301806) * (alpha - 2) * (alpha - 1) * (alpha + 1/2) * (alpha + 1)^3]

In [5]:
dopTardigrade5ptd4case3desing=dopTardigrade5ptd4case3.desingularize()

In [6]:
dopTardigrade5ptd4case3desing.leading_coefficient().factor()

(z - 4) * (z + 1/2)^3 * z^4 * (z^2 - z - 1/4) * (z^3 - 64/25*z^2 - 117/50*z + 81/100)

In [15]:
s = open("PF-Quartic-case3.txt").readlines()[0]
dopQuarticcase3 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopQuarticcase3.order()]

[6]

In [7]:
dopQuarticcase3desing=dopQuarticcase3.desingularize()

In [8]:
dopQuarticcase3desing.leading_coefficient().factor()

(z + 1/2)^3 * (z - 4)^5 * z^6 * (z^2 - z - 1/4) * (z^3 - 64/25*z^2 - 117/50*z + 81/100)

The factorisation is in the text file <a href="PF-Tardigrade-5pt-factor.txt">PF-Tardigrade-5pt-factor.txt</a>

In [ ]:
#fac=dopTardigrade5ptd4case4.factor(verbose=True); 
#[x.order() for x in fac]
#with open('PF-Tardigrade-5pt-case4-factor.txt',"w") as f:
#     f.write(str(fac))

In [5]:
file = open('PF-Tardigrade-5pt-case4-factor.txt', 'r')
lines = file.readlines()
s = ''.join(lines);
fac1=eval(preparse(s))
[x.order() for x  in fac1]
# this is irreducible

[6]

<b>the case with one massless internal mass</b><br>
``F := (64*x1 + 81*x2 + 9*x3 + 25*x4 + 74*x6)*((x3 + x4)*(x5 + x6) + x1*(x3 + x4 + x5 + x6) + x2*(x3 + x4 + x5 + x6)) - t*(784*x3*x5*x6 + 784*x4*x5*x6 + 4*x3*x4*(x5 + x6) + x1*(4*x3*x4 + 289*x3*x5 + 1783*x4*x5 + 1525*x3*x6 + 2274*x4*x6 + 784*x5*x6 + 1521*x2*(x3 + x4 + x5 + x6)) + x2*(446*x4*x6 + 784*x5*x6 + x3*(4*x4 + 25*x5 + 1216*x6)))``
<br>
The Picard-Fuchs differential operator is given in <a href="PF-Tardigrade-5pt-d4-case5.txt">PF-Tardigrade-5pt-d4-case5.txt</a>

In [16]:
s = open("PF-Tardigrade-5pt-d4-case5.txt").readlines()[0]
dopTardigrade5ptd4case5 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopTardigrade5ptd4case5.order()]

[11]

In [18]:
dopTardigrade5ptd4case5.leading_coefficient().factor()

(z - 16) * (z - 1) * (z - 289/1521) * (z - 1/1521) * z^3 * (z^24 - 20701378363912619192162426051901648508471589253708755904899042341096613109435495741332630704980488417/257075932666088337868562784693383455303174056378730311758000872766554447833638685248569436942467770*z^23 + 846872270925911628277108346100404467614349306622044981037067984052940175921302002863257364411536328019908437/340615327745260404142330947207345342938493497739562513866880836380773981201657912406944561171292096539200*z^22 - 59736755197563449450587099169620782810948137889295986712477950423338881720329744473369213601408280696815704979012863/1523143185691590759619426989864974463831778913581911275759085711277362242698701753226630271972113719778202208000*z^21 + 39510649436705312082280208611995410183208537515414988723268700588771842056014868057716436216144063672734585026554467186214461/108977604946952225014729223162540814542321904624332414852206943016758909362647881247578426438217276884495647065630720000*z^20 - 36357443280

In [17]:
s = open("PF-Quartic-case5.txt").readlines()[0]
dopQuarticcase5 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopQuarticcase5.order()]

[11]

In [19]:
dopQuarticcase5.leading_coefficient().factor()

(z - 16) * (z - 1) * z^3 * (z - 289/1521)^10 * (z - 1/1521)^10 * (z^24 - 20701378363912619192162426051901648508471589253708755904899042341096613109435495741332630704980488417/257075932666088337868562784693383455303174056378730311758000872766554447833638685248569436942467770*z^23 + 846872270925911628277108346100404467614349306622044981037067984052940175921302002863257364411536328019908437/340615327745260404142330947207345342938493497739562513866880836380773981201657912406944561171292096539200*z^22 - 59736755197563449450587099169620782810948137889295986712477950423338881720329744473369213601408280696815704979012863/1523143185691590759619426989864974463831778913581911275759085711277362242698701753226630271972113719778202208000*z^21 + 39510649436705312082280208611995410183208537515414988723268700588771842056014868057716436216144063672734585026554467186214461/108977604946952225014729223162540814542321904624332414852206943016758909362647881247578426438217276884495647065630720000*z^20 - 36357

<b> the case with one massless internal mass</b><br>
``F := (x1 + x2 + x3 + x4 + 2*x6)*((x3 + x4)*(x5 + x6) + x1*(x3 + x4 + x5 + x6) + x2*(x3 + x4 + x5 + x6)) - t*(x3*x4*x5 + x2*x3*(x4 + x5) + x3*x4*x6 + x3*x5*x6 + x4*x5*x6 + x2*(-2*x4 + x5)*x6 + x1*(x5*(x4 + x6) + x3*(x4 + x5 + x6) + x2*(x3 + x4 + x5 + x6))``

In [7]:
s = open("PF-Tardigrade-5pt-d4-case6.txt").readlines()[0]
dopTardigrade5ptd4case6 = OA(s.replace("^", "**").replace("t", "z").replace("D", "Dz"))
[dopTardigrade5ptd4case6.order(),dopTardigrade5ptd4case6.degree(),dopTardigrade5ptd4case6.indicial_polynomial(z).factor()]

[6, 50, (-15443427328) * (alpha - 1) * alpha * (alpha + 1/2) * (alpha + 1)^3]